src/style.rs

```diff
@@ -0,0 +1,162 @@
+use std::borrow::Cow;
+
+/// Rich (formatted) text cell.
+#[derive(Debug, PartialEq, Clone, Default)]
+pub struct RichText {
+    /// Total text, where the different formats are saved in the `formatted` field.
+    text: String,
+    /// List of format annotations, with each having:
+    /// - Length of string section
+    /// - Format of the string section
+    formatted: Vec<(usize, FontFormat)>,
+}
+
+impl RichText {
+    /// Create new empty rich text.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Create an instance with only default plain text.
+    pub fn plain(s: String) -> Self {
+        let len = s.len();
+        Self {
+            text: s,
+            formatted: vec![(len, FontFormat::default())],
+        }
+    }
+
+    /// Is this empty?
+    #[inline]
+    pub fn is_empty(&self) -> bool {
+        self.text.is_empty()
+    }
+
+    /// Is this just plain text?
+    #[inline]
+    pub fn is_plain(&self) -> bool {
+        self.formatted.iter().all(|(_len, f)| f.is_default())
+    }
+
+    /// Get the full text value without formatting.
+    #[inline]
+    pub fn text(&self) -> &String {
+        &self.text
+    }
+
+    /// Deconstruct this into a plain string with the text.
+    #[inline]
+    pub fn into_text(self) -> String {
+        self.text
+    }
+
+    /// Add a formatted text element to the rich text.
+    #[inline]
+    pub fn add_element(&mut self, element: RichTextPart<'_>) -> &mut Self {
```

Comment:
Not very important but I'd expect `add_xxx`, `push_xxx` and `append_xxx` to return an element that would be evicted or an index. 
Here this is more like a builder pattern so a `with_element` seems better.

Reply:
I think I can easily return the index and have it be a push, as it really behaves more like a push.

Reply:
Renamed to `push` and returns nothing, similar to `Vec::push`.